apps/stockspotlight/stock-spotlight.star

```diff
@@ -0,0 +1,197 @@
+"""
+Applet: Stock Spotlight
+Summary: Showcase 3 Stocks
+Description: Showcase up to 3 of your favorite stocks using your own Finnhub API key (available from https://finnhub.io).
+Author: Seth Cottle
+"""
+
+load("cache.star", "cache")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+
+CACHE_TTL = 60  # Cache for 1 minute
+DISPLAY_TIME = 5  # Display each stock for 5 seconds
+FINNHUB_GREEN = "#1DB954"
+
+def main(config):
+    api_key = config.get("api_key", "")
+    if not api_key or len(api_key) < 10:  # Basic check for API key
+        return render_config_screen("Please enter a valid Finnhub API key")
+
+    stocks = [config.get("stock{}".format(i), "").upper() for i in range(1, 4) if config.get("stock{}".format(i))]
+    if not stocks:
+        return render_config_screen("Please enter at least one stock symbol")
+
+    stock_data = get_stock_data(api_key, stocks)
+
+    if not stock_data:
+        return render_config_screen("Unable to fetch stock data. Please check your API key and internet connection.")
+
+    return render.Root(
+        child = render.Animation(
+            children = [create_stock_display(data) for data in stock_data if data],
+        ),
+        delay = DISPLAY_TIME * 1000,
+    )
+
+def render_config_screen(message):
+    return render.Root(
+        child = render.Box(
+            render.Column(
+                expanded = True,
+                main_align = "space_between",
+                cross_align = "center",
+                children = [
+                    render.Padding(
+                        pad = (0, 2, 0, 0),
+                        child = render.Text("FINNHUB", color = FINNHUB_GREEN, font = "6x13"),
+                    ),
+                    render.Marquee(
+                        width = 64,
+                        child = render.Text(message, color = FINNHUB_GREEN),
+                        offset_start = 64,
+                        offset_end = 64,
+                    ),
+                    render.Box(width = 64, height = 1),
+                ],
+            ),
+        ),
+    )
+
+def get_stock_data(api_key, stocks):
+    all_stock_data = []
+    for symbol in stocks:
+        if len(symbol) < 1 or len(symbol) > 5:  # Basic validation for stock symbol length
+            continue
+        cached_data = cache.get("stock_data_" + symbol)
```

Comment:
You should be able to simplify caching a bit by relying on the http module's built-in cache functionality. Just pass `ttl_seconds` to `http.get()`. Would be great to see that change before merging!